docs/internals/contributing/writing-code/accessibility.txt

```diff
@@ -0,0 +1,114 @@
+=============
+Accessibility
+=============
+
+All aspects of Django must be compatible with our accessibility standards,
+and with the specific support targets we aim for.
+
+.. _accessibility-standards:
+
+Accessibility standards
+=======================
+
+Django aims for conformance with `WCAG 2.2`_ at the AA level.
+WCAG is the most well established standard for web accessibility. AA-level
+guidelines are the most common legal compliance target around the world.
+
+Wherever applicable, we also attempt to follow other best practices, such as:
+
+* WCAG 2.2 AAA-level guidelines.
+* The upcoming `WCAG 3.0`_ guidelines.
+* `ATAG 2.0`_ guidelines for authoring tools.
+
+.. note::
+    Do you want to learn more about accessibility without diving straight into
+    standards? We recommend `The A11Y Project`_,
+    a community-driven effort to make digital accessibility easier.
+
+.. _WCAG 2.2: https://www.w3.org/TR/WCAG22/
+.. _WCAG 3.0: https://www.w3.org/TR/wcag-3.0/
+.. _ATAG 2.0: https://www.w3.org/TR/ATAG20/
+.. _The A11Y Project: https://www.a11yproject.com/
+
+.. _accessibility-support-targets:
+
+Accessibility support targets
+=============================
+
+Beyond standards, we want to make sure Django actually works for people using
+a wide range of input devices and assistive technologies.
+The best way to do this effectively is to take accessibility considerations
+into account as part of designing features. If in doubt, consult
+with users who rely on assistive technologies or with accessibility experts.
+
+.. _accessibility-testing-baseline:
+
+Accessibility testing baseline
+------------------------------
+
+ Always test user interface changes with:
+
+* Keyboard-only navigation
+* Automated checks in the `Accessibility Insights`_ browser extension,
+  or an equivalent tool with the `Axe`_ checker.
+
+Where appropriate, also test with:
+
+* Touch-only navigation
+* 400% browser zoom
+* forced-colors mode DevTools emulation
+
+.. note::
+    How much should you test? Devise the UI with accessibility in mind,
+    and the testing will only be needed as a final check. For more complex
+    interfaces, confer with other contributors to decide on testing targets.
+
+.. _assistive-technologies:
+
+Assistive technologies
+----------------------
+
+Where more extensive testing is appropriate,
+here are free assistive technologies we recommend testing with:
+
+* `NVDA`_ on Windows with Firefox ESR
+* `VoiceOver`_ on macOS with Safari
+* `Narrator`_ on Windows with Microsoft Edge
+* Windows Magnifier and macOS Zoom
+* Windows Speech Recognition and macOS Voice Control
+* VoiceOver on iOS, or TalkBack on Android
+* `Contrast themes`_ on Windows
```

Comment:
This is heavily based on https://docs.wagtail.org/en/stable/contributing/developing.html#accessibility-targets / https://www.gov.uk/service-manual/technology/testing-with-assistive-technologies.

For the more demanding changes we would want to test with _all_ of those things, but is burdensome due to requiring 4 different operating systems. At least there’s something for everyone!

Worth mentioning two possible options I chose to exclude for now:

- [Orca](https://help.gnome.org/users/orca/stable/index.html.en) for Linux. There’s 0 reported usage on [screen reader user surveys](https://webaim.org/projects/screenreadersurvey9/) and 0.5% reported on https://accessibility.blog.gov.uk/2016/11/01/results-of-the-2016-gov-uk-assistive-technology-survey/. I’ve never used it myself nor know anyone who has used it. I think it would be a good one to include _if_ we have anyone contributing in Django who is willing to test with it.
- [ChromeVox](https://support.google.com/chromebook/answer/7031755?hl=en-GB) on ChromeOS. There is some reported usage in survey but as far as I know there is no other way to test it than having a ChromeBook (the browser extension for Chrome isn’t a viable option). Again a good one to include _if_ we have anyone contributing in Django who is willing to test with it.

Reply:
> Worth mentioning two possible options I chose to exclude for now:
> * [Orca](https://help.gnome.org/users/orca/stable/index.html.en) for Linux. There’s 0 reported usage on [screen reader user surveys](https://webaim.org/projects/screenreadersurvey9/) and 0.5% reported on https://accessibility.blog.gov.uk/2016/11/01/results-of-the-2016-gov-uk-assistive-technology-survey/. I’ve never used it myself nor know anyone who has used it. I think it would be a good one to include _if_ we have anyone contributing in Django who is willing to test with it.

I believe it is important to also include Orca for Linux in the list of technologies for many reasons, including:
- in the WebAIM survey it is specified that Orca is present among the technologies but has been merged into "others" (see [WebAIM: Screen Reader User Survey #5 Results](https://webaim.org/projects/screenreadersurvey5/#primary) and [WebAIM: Screen Reader User Survey #6 Results](https://webaim.org/projects/screenreadersurvey6/#primary) _"While they were not listed as response options for simplicity's sake, the survey comments indicate that Orca, SuperNova, and Speakup (among others) were common "Other" screen readers."_)
- Linux is the most used operating system among Django developers with 41% (see [Django Developers Survey 2022 Results](https://lp.jetbrains.com/django-developer-survey-2022/#circular-chart-685)), which makes them a decidedly different user base than that of the aforementioned survey, where it is only 1.4% (see [WebAIM: Screen Reader User Survey #9 Results](https://webaim.org/projects/screenreadersurvey9/#os))
- it could be an excellent opportunity to enrich future surveys for Django, Python or even Wgatail by asking actual developers what assistive technologies they use
- I have only tried to use Orca, but if we don't find other users of it in the accessibility team I could help you with this activity

Reply:
Thanks @pauloxnet! Those sound like excellent reasons to me, I’ll add it unless others raise objections.